x-pack/plugin/profiling/src/main/java/org/elasticsearch/xpack/profiling/GetStackTracesRequest.java

```diff
@@ -330,7 +350,7 @@ public String[] indices() {
         if (this.indices == null) {
             indices.addAll(EventsIndex.indexNames());
         } else {
-            indices.add(this.indices);
+            indices.addAll(List.of(this.indices));
         }
         return indices.toArray(new String[0]);
```

Comment:
Maybe use pre-allocation here as well?

Reply:
That's actually the preferred style in Java, see also https://github.com/JetBrains/intellij-community/blob/master/java/java-impl/src/inspectionDescriptions/ToArrayCallWithZeroLengthArrayArgument.html for the rationale.